Update CI jobs

Kichura opened the pull request:
1. Updates checkout, setup-java and setup-gradle to v4,
2. Replaces the old Gradle job *(`gradle/gradle-build-action`)* with new variant, *(`gradle/actions/setup-gradle`)*
~~3. Replaces the old validation job *(`gradle/wrapper-validation-action`)* with new variant. *(`gradle/actions/wrapper-validation`)* Removes the validation job in favor of setup-gradle,~~
4. Sets `persist-credentials` to false to prevent exposing to greater risks than imagined.

Kichura committed:
Update CI jobs.

Kichura committed:
Replace old gradle job with new one.

Kichura committed:
Replace old validation action with new one.

jpenilla commented:
wrapper-validation action is no longer needed, setup-gradle v4 validates the wrapper

Kichura commented:
> wrapper-validation action is no longer needed, setup-gradle v4 validates the wrapper

~~will remove.~~ removed.

Kichura committed:
Remove validation action.

This was requested by jpenilla.

jpenilla commented on .github/workflows/build.yml:
```diff
@@ -12,13 +12,11 @@ jobs:
+        uses: actions/checkout@v4
       - name: Setup Gradle
-        uses: gradle/gradle-build-action@v2
+        uses: gradle/actions/setup-gradle@v3
```
v4

Kichura replied:
Skipped this earlier cuz origin repo did not reflect the v4 update in readme.

Kichura committed:
Bump setup-gradle to v4.

Also requested by jpenilla.

Kichura committed:
Do not persist credentials.

Prevents exposing the workflow to greater security risks than imagined, This was spotted by the "zizmor" rust library.

kennytv commented:
Sorry about that - can you reopen a PR targeting the "new" master branch?

Kichura commented:
> Sorry about that - can you reopen a PR targeting the "new" master branch?

will be reopened under new branch.
